ffi/src/error.rs

```diff
@@ -93,6 +94,32 @@ impl From<WrongOSError> for IronRdpErrorKind {
     }
 }
 
+impl From<der::Error> for IronRdpErrorKind {
+    fn from(_val: der::Error) -> Self {
+        IronRdpErrorKind::DecodeError
+    }
+}
+
+impl From<ironrdp_rdcleanpath::MissingRDCleanPathField> for IronRdpErrorKind {
+    fn from(_val: ironrdp_rdcleanpath::MissingRDCleanPathField) -> Self {
+        IronRdpErrorKind::Generic
+    }
+}
+
+pub struct GenericError(pub String);
+
+impl Display for GenericError {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl From<GenericError> for IronRdpErrorKind {
+    fn from(_val: GenericError) -> Self {
+        IronRdpErrorKind::Generic
+    }
+}
```

Comment:
suggestion: Use `anyhow::Error` instead, so you can use `anyhow::Context` methods. Then use `let repr = format!("{error:#}")` to properly retrieve the full error message.

note: Similarly, for IronRDP errors, you may need to use the error reporter in order to obtain the full trace. I see that’s not what we do today.